src/textual/widgets/text_input.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from rich.console import RenderableType
+from rich.text import Text
+
+from textual import events
+from textual._types import MessageTarget
+from textual.geometry import Size
+from textual.keys import Keys
+from textual.message import Message
+from textual.reactive import Reactive
+from textual.widget import Widget
+
+
+class TextInputBase(Widget):
+    ALLOW_PROPAGATE = {"tab", "shift+tab"}
```

Comment:
This probably belongs in the existing child class since for more advanced editors we don't want these to propagate.